src/transforms/remap.rs

```diff
@@ -169,6 +173,8 @@ impl FunctionTransform for Remap {
 
                 if !self.drop_on_error {
                     output.push(original_event.expect("event will be set"))
+                } else {
+                    err_output.push(original_event.expect("event will be set"))
                 }
```

Comment:
Piggybacking on this also feels weird, but it is rather simple and has the one large advantage of being both backwards compatible and not introducing any new knobs. Very open to any other suggestions.

Reply:
I'm not opposed to this solution, as I think we'll be tweaking `drop_on_error` and `drop_on_abort` in the future regardless (see https://github.com/vectordotdev/vector/issues/7496).

The one problem I see with this is that you can't use VRL's `abort` keyword in this case to reroute. Perhaps we should also add an `aborts` named output for aborted programs?

The other downside to this, is that there's like a (minor?) performance drop if you just want to drop events and never read from the errors/aborts channels. It would be nice if we could somehow short-circuit sending events over those channels if there's no need to, without having the user have to specify that through a config flag (e.g. we'd need the topology to somehow expose a list of components listening on those specific inputs).

But, that last point isn't blocking to me, and it seems that `aborts` is an easy one to add, if we agree on that.